src/algebra/order/to_interval_mod.lean

```diff
@@ -457,6 +458,89 @@ begin
     rw [hz, to_Ioc_mod_zsmul_add] }
 end
 
+section Ico_Ioc
+
+variables (a : α) {b : α} (hb : 0 < b) (x : α)
+
+omit hα
+/-- `mem_Ioo_mod a b x` means that `x` lies in the open interval `(a, a + b)` modulo `b`.
+Equivalently (as shown below), `x` is not congruent to `a` modulo `b`, or `to_Ico_mod a hb` agrees
+with `to_Ioc_mod a hb` at `x`, or `to_Ico_div a hb` agrees with `to_Ioc_div a hb` at `x`. -/
+def mem_Ioo_mod (b x : α) : Prop := ∃ z : ℤ, x + z • b ∈ set.Ioo a (a + b)
+include hα
+
+lemma tfae_mem_Ioo_mod :
+  tfae [mem_Ioo_mod a b x,
+    to_Ico_mod a hb x = to_Ioc_mod a hb x,
+    to_Ico_mod a hb x + b ≠ to_Ioc_mod a hb x,
+    to_Ico_mod a hb x ≠ a] :=
+begin
```

Comment:
Sorry for the delayed review; this looks along the lines of what I was thinking of; thanks!